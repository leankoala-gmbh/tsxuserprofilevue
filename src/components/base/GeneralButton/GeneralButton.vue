<script lang="ts" setup>
type TButtonTags = 'a' | 'button' | 'div'
type TVariant = 'regular' | 'ghost' | 'transparent' | 'cancel'

const emit = defineEmits(['click'])

const props = defineProps({
  tag: {
    type: String as () => TButtonTags,
    default: 'button'
  },
  isDisabled: {
    type: Boolean,
    default: false
  },
  variant: {
    type: String as () => TVariant,
    default: 'regular'
  }
})

const sendClick = () => {
  if (props.isDisabled) return
  emit('click')
}
</script>

<template>
  <component
    :is="tag"
    :disabled="isDisabled"
    class="rounded h-10 flex items-center justify-center px-4 duration-300 transition border-current border font-medium"
    :class="[
      { 'isDisabled': isDisabled },
      { 'hover:underline': variant === 'transparent'},
      `generalButton--${variant}`
    ]"
    @click.stop="sendClick"
  >
    <slot />
  </component>
</template>

