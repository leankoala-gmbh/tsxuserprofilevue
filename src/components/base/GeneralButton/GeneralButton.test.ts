import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import GeneralButton from './GeneralButton.vue'

describe('GeneralButton.vue', () => {
  it('check snapshot', () => {
    const wrapper = mount(GeneralButton)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
