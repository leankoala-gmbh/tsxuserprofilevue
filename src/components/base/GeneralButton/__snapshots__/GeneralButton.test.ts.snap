// Vitest Snapshot v1

exports[`GeneralButton.vue > check snapshot 1`] = `"<button class=\\"rounded h-10 flex items-center justify-center px-4 duration-300 transition border-current border font-medium generalButton--regular\\"></button>"`;
